tap_rest_api_msdk/tap.py

```diff
@@ -482,26 +484,33 @@ def get_schema(
             A schema for the stream.
 
         """
-        # todo: this request format is not very robust
+        # TODO: this request format is not very robust
+
+        # Initialise Variables
+        auth_method = self.config.get("auth_method", "")
+        self.http_auth = None
 
-        auth_method = self.config.get('auth_method', '')
-        if auth_method and not auth_method == 'no_auth':
-            # Initializing Authenticator required in TAP for to dynamically discover schema
-            authenticator = self.authenticator
-            if authenticator:
+        if auth_method and not auth_method == "no_auth":
+            # Initializing Authenticator for authorisation to obtain a schema.
+            # Will set the self.http_auth if required by a given authenticator
+            authenticator = select_authenticator(self)
+            if hasattr(authenticator, "auth_headers"):
                 headers.update(authenticator.auth_headers or {})
+            if hasattr(authenticator, "auth_params"):
                 params.update(authenticator.auth_params or {})
 
-        r = requests.get(self.config["api_url"] + path, params=params, headers=headers)
+        r = requests.get(self.config["api_url"] + path, auth=self.http_auth, params=params, headers=headers)
```

Comment:
Noted in review meeting, `self.http_auth` is set to `None` and then becomes populated in the `if` block below. Also some more thought in a future release about get functions in `authenticator` so that the two `if` statements can be condensed